### PR TITLE
Add Global and ThreadLocal structure

### PR DESCRIPTION
**Pull request by clover2123**
* Platform is also managed in Global object
* Global has life time same to program
* ThreadLocal manage global values allocated for each thread
* Update PlatformRef to allow users to add thread-local custom data

Signed-off-by: HyukWoo Park <hyukwoo.park@samsung.com>

### CODE REVIEW

**Comment by ksh8281** [comment]
We cannot share Platform class like this.
because Platform has loadedModule data.
If you want to share Platform across VMInstace or threads. we should divide it's data per Platform

**Comment by ksh8281** [comment]
IHMO choose some thread into main thread is not good.
because there is no huge difference with main-sub thread.
and I think main thread is ended before sub thread end

**Comment by bbrto21** [comment]
@clover2123 please resolve a conflict

**Comment by clover2123** [comment]
@ksh8281 
I updated a patch to cover your review.
`ThreadLocal` is newly added to manage all thread-global values.
So, `Global` is shared between threads while `ThreadLocal` is thread specific.

File: src/api/EscargotPublic.cpp [hunk 1]
```diff
@@ -102,24 +105,175 @@ inline AtomicString toImpl(AtomicStringRef* v)
...
+
+    virtual void* onReallocArrayBufferObjectDataBuffer(void* oldBuffer, size_t oldSizeInByte, size_t newSizeInByte) override
+    {
+        return m_platform->onReallocArrayBufferObjectDataBuffer(oldBuffer, oldSizeInByte, newSizeInByte);
+    }
+
+    virtual void markJSJobEnqueued(Context* relatedContext) override
+    {
+        // Job should be enqueued in the main thread
+        ASSERT(Global::inMainThread());
```
Comment (ksh8281) [inline src/api/EscargotPublic.cpp:139]: IMHO we don't need to specify main-thread
Every thread can be act as main-thread.
And worker threads can use promise too

**Review by ksh8281 — APPROVED** [review]

File: src/api/EscargotPublic.cpp (hunk 1 above)
Comment (clover2123) [reply src/api/EscargotPublic.cpp]: Fixed it.
Regarding job queue, I'll separate it in later patch.

**Review by bbrto21 — APPROVED** [review]

**Pull request merged**